script/tool/lib/src/license_check_command.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:path/path.dart' as p;
+
+import 'common.dart';
+
+const Set<String> _codeFileExtensions = <String>{
+  '.c',
+  '.cc',
+  '.cpp',
+  '.dart',
+  '.h',
+  '.java',
+  '.m',
+  '.mm',
+  '.swift',
+  '.sh',
+};
+
+// Basenames without extensions of files to ignore.
+const Set<String> _ignoreBasenameList = <String>{
+  'flutter_export_environment',
+  'GeneratedPluginRegistrant',
+  'generated_plugin_registrant',
+};
+
+// File suffixes that otherwise match _codeFileExtensions to ignore.
+const Set<String> _ignoreSuffixList = <String>{
+  '.g.dart', // Generated API code.
+  '.mocks.dart', // Generated by Mockito.
+};
+
+// Full basenames of files to ignore.
+const Set<String> _ignoredFullBasenameList = <String>{
+  'resource.h', // Generated by VS.
+};
+
+// Copyright and license regexes.
+//
+// These are intentionally very simple, since almost all source in this
+// repository should be using the same license text, comment style, etc., so
+// they shouldn't need to be very flexible. Complexity can be added as-needed
+// on a case-by-case basis.
+final RegExp _copyrightRegex = RegExp(r'^(?://|#) Copyright', multiLine: true);
+// All Flutter-authored code.
+final RegExp _bsdLicenseRegex = RegExp(
+    r'^(?://|#) Use of this source code is governed by a BSD-style license',
+    multiLine: true);
+// Other code. When adding license regexes here, include the copyright info to
+// ensure that any new additions are flagged for added scrutiny in review.
+// -----
+// Third-party code used in url_launcher_web.
+final RegExp _workivaLicenseRegex = RegExp(
```

Comment:
We should enforce that all code that doesn't have our copyright notice is in a third_party directory.

Reply:
Done.